components/script/dom/blob.rs

```diff
@@ -223,6 +230,66 @@ impl BlobMethods for Blob {
         let blob_impl = BlobImpl::new_sliced(rel_pos, self.blob_id.clone(), type_string);
         Blob::new(&*self.global(), blob_impl)
     }
+
+    // https://w3c.github.io/FileAPI/#text-method-algo
+    fn Text(&self) -> Rc<Promise> {
+        let global = self.global();
+        let in_realm_proof = AlreadyInRealm::assert(&global);
+        let p = Promise::new_in_current_realm(&global, InRealm::Already(&in_realm_proof));
+        let id = self.get_blob_url_id();
+        global.read_file_async(
+            id,
+            p.clone(),
+            Box::new(|promise, bytes| match bytes {
+                Ok(b) => {
+                    let (text, _, _) = UTF_8.decode(&b);
```

Comment:
Here I am wondering if we can re-use [`run_text_data_algorithm`](https://github.com/servo/servo/blob/12693b51f510761d5dd58a9103596a192edb803c/components/script/body.rs#L128)

Reply:
Ah no forget about this one, I think what you are doing with UTF_8 fits the spec more, in the light of:

> Note: This is different from the behavior of readAsText() to align better with the behavior of Fetch’s text(). Specifically this method will always use UTF-8 as encoding, while FileReader can use a different encoding depending on the blob’s type and passed in encoding name. 
https://w3c.github.io/FileAPI/#text-method-algo

I think the other comment is still relevant.